src/Language/PureScript/AST/Declarations.hs

```diff
@@ -29,6 +29,9 @@ import qualified Text.Parsec as P
 -- | A map of locally-bound names in scope.
 type Context = [(Ident, Type)]
 
+data TypeSearch = TSBefore Environment | TSAfter [(Qualified Ident, Type)]
```

Comment:
Could you please break this over a couple of lines and add some Haddock comments?